app/celery/letters_pdf_tasks.py

```diff
@@ -157,24 +163,26 @@ def letter_in_created_state(filename):
 @notify_celery.task(name='process-virus-scan-passed')
 def process_virus_scan_passed(filename):
     current_app.logger.info('Virus scan passed: {}'.format(filename))
-    move_scanned_pdf_to_letters_pdf_bucket(filename)
     reference = get_reference_from_filename(filename)
-    updated_count = update_letter_pdf_status(reference, NOTIFICATION_CREATED)
+    notification = dao_get_notification_by_reference(reference)
 
-    if updated_count != 1:
-        raise Exception(
-            "There should only be one letter notification for each reference. Found {} notifications".format(
-                updated_count
-            )
-        )
+    is_test_key = notification.key_type == KEY_TYPE_TEST
+    move_scanned_pdf_to_test_or_live_pdf_bucket(
+        filename,
+        is_test_letter=is_test_key
+    )
+    update_letter_pdf_status(
+        reference,
+        NOTIFICATION_DELIVERED if is_test_key else NOTIFICATION_CREATED
+    )
 
 
 @notify_celery.task(name='process-virus-scan-failed')
 def process_virus_scan_failed(filename):
     current_app.logger.error('Virus scan failed: {}'.format(filename))
```

Comment:
I think we want this to be an exception in the first instance

Reply:
ok